### PR TITLE
liquibase: include all jars from internal/lib instead of just a few

### PR DESCRIPTION
**Pull request by agilesteel**
###### Description of changes

The official installer installs more jars by default than this nix installer. Multiple commands don't work as intended because of that. In fact even `liquibase init project` - a command everyone would start with is broken. Furthermore since the `snakeyaml.jar` is not included it is impossible to parse `root.yml` files.

This PR adds all jars and also moves them into the `internal/lib` directory - the same as the default installer would.

I also tried aligning the output of the `liquibase --version` command, but even though I moved the jars into the `internal/lib` folder they are not listed. They are on the classpath though, which is also why the aforementioned commands work as intended. It's just that they are not listed when calling the `liquibase --version` command. I can live with that...

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by agilesteel** [comment]
Hey @jsoo1 and @AndersonTorres!

I've never contributed to `liquibase` before and you were the last ones to touch it. Could you please review this PR? 🙏🏼 Thanks

**Comment by jsoo1** [comment]
LGTM, cc @ptliu-awake

File: pkgs/development/tools/database/liquibase/default.nix [hunk 1]
```diff
@@ -48,7 +48,10 @@ stdenv.mkDerivation rec {
     in
     ''
       mkdir -p $out
-      mv ./{lib,licenses,internal/lib/liquibase-core.jar,internal/lib/postgresql.jar,internal/lib/picocli.jar} $out/
+      mv ./{lib,licenses} $out/
+
+      mkdir -p $out/internal/lib
+      mv ./internal/lib/*.jar $out/internal/lib/
```
Comment (jsoo1) [inline pkgs/development/tools/database/liquibase/default.nix:54]: Only jars?

File: pkgs/development/tools/database/liquibase/default.nix [hunk 2]
```diff
@@ -63,12 +66,12 @@ stdenv.mkDerivation rec {
...
       # taken from the executable script in the source
       CP=""
+      ${addJars "$out/internal/lib"}
       ${addJars "$out/lib"}
       ${addJars "$out"}
       ${lib.concatStringsSep "\n" (map (p: addJars "${p}/share/java") extraJars)}
-
       ${lib.getBin jre}/bin/java -cp "\$CP" \$JAVA_OPTS \
-        liquibase.integration.commandline.LiquibaseCommandLine \''${1+"\$@"}
+      liquibase.integration.commandline.LiquibaseCommandLine \''${1+"\$@"}
```
Comment (jsoo1) [inline pkgs/development/tools/database/liquibase/default.nix:74]: @ptliu-awake Is this the correct invocation?

**Review by jsoo1 — APPROVED** [review]
LGTM but @ptliu-awake will know better than I.

File: pkgs/development/tools/database/liquibase/default.nix (hunk 1 above)
Comment (agilesteel) [reply pkgs/development/tools/database/liquibase/default.nix:54]: I tried to keep it as close to the original code as possible. There are only jars in that directory anyways. I could change it though if you think it's better. I don't mind.

File: pkgs/development/tools/database/liquibase/default.nix (hunk 2 above)
Comment (agilesteel) [reply pkgs/development/tools/database/liquibase/default.nix:74]: The original indentation produced 3 spaces in the run script as opposed to 1.

Before:

```bash
/nix/store/dvad9y3g9sd4kw3l2i1qk5s7dsg6w009-graalvm11-ce-22.3.1/bin/java -cp "$CP" $JAVA_OPTS   liquibase.integration.commandline.LiquibaseCommandLine ${1+"$@"}
```

After:

```bash
/nix/store/dvad9y3g9sd4kw3l2i1qk5s7dsg6w009-graalvm11-ce-22.3.1/bin/java -cp "$CP" $JAVA_OPTS liquibase.integration.commandline.LiquibaseCommandLine ${1+"$@"}
```

**Comment by agilesteel** [comment]
Can we merge this one please?

**Comment by jsoo1** [comment]
I approve of merging but I don't have a commit bit, apologies.

**Comment by kubukoz** [comment]
Result of `nixpkgs-review pr 242696` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>liquibase</li>
  </ul>
</details>

**Pull request merged**